python/tk_multi_publish2/api/data.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2017 Shotgun Software Inc.
```

Comment:
This existed somewhat before here: https://github.com/shotgunsoftware/tk-multi-publish2/blob/master/python/tk_multi_publish2/processing/item.py#L646

It's now the base class for item properties and for hook/plugin settings.